### PR TITLE
domain: add BlockCopy() method

### PR DESCRIPTION
**Pull request by kunteynir**

### CODE REVIEW

**Comment by rmohr** [comment]
Hi @kunteynir,

 * could you move the BlockCopy code from domain.go into domain_1.2.14.go? I think it requires at least 1.2.6.
 * do you see a way to add a test for this method?


@vincentbernat do we want to have additional tags to 1.2.14?

**Comment by vincentbernat** [comment]
I don't think we want additional tags if nobody request them.

**Comment by kunteynir** [comment]
Hi @rmohr, I moved BlockCopy code from domain.go into domain_1.2.14.go in 793813dafb227bee8c4b4d0f064e157b08ab0c33 and would try to implement test for BlockCopy method in next PR, where I'd like to add BlockJobAbort and GetBlockJobInfo domain methods.

**Comment by rmohr** [comment]
I would prefer if you would add the tests before adding the other stuff, but great work :+1:

**Comment by kunteynir** [comment]
Hi @rmohr, I added one common test for all methods together in a03bb4ed12d2e455cd8f07796d26e42642714893 because there is no way to do it separately.

File: domain_1.2.14.go [hunk 1]
```diff
@@ -70,3 +71,82 @@ func (d *VirDomain) ListAllInterfaceAddresses(src uint) ([]VirDomainInterface, e
...
+
+	for _, param := range *dest {
+
+		switch param.Name {
+
+		case VIR_DOMAIN_BLOCK_COPY_GRANULARITY:
+			value, ok := param.Value.(uint)
+			if !ok {
+				err = errors.New("Mismatched parameter value type")
+
```
Comment (rmohr) [inline domain_1.2.14.go]: We don't have this empty line between the code and return statements. Could you remove all newlines before the return statements?

File: domain_1.2.14.go [hunk 2]
```diff
@@ -70,3 +71,82 @@ func (d *VirDomain) ListAllInterfaceAddresses(src uint) ([]VirDomainInterface, e
...
+			cName := C.CString(param.Name)
+			defer C.free(unsafe.Pointer(cName))
+
+			result := C.virTypedParamsAddUInt(&params, &nParams, &maxParams, cName, C.uint(value))
+			if result == -1 {
+				err = GetLastError()
+
+				return
+			}
+
```
Comment (rmohr) [inline domain_1.2.14.go]: Could you remove the empty lines in the switch statement between the cases?

File: domain_1.2.14_test.go [hunk 3]
```diff
@@ -28,3 +28,110 @@ func TestDomainListAllInterfaceAddresses(t *testing.T) {
...
+		</domain>`,
+		VIR_DOMAIN_NONE)
+	if err != nil {
+		t.Fatalf("DomainCreateXML: %s", err.Error())
+	}
+	defer dom.Free()
+	defer dom.Destroy()
+
+	var (
+		stop                     = make(chan struct{})
```
Comment (rmohr) [inline domain_1.2.14_test.go]: I would prefer one space between the variable name and the equal sign

**Comment by kunteynir** [comment]
@rmohr, it is ok now.

**Comment by rmohr** [comment]
thx again. :+1:

**Pull request merged**